// giao-cdn/giao-plugin-config.ts
import { GiaoPluginConfig, OptionType } from 'giao';

const config: GiaoPluginConfig = {
    desc: '上传静态资源至快手cdn',
    commandsDir: 'lib/commands',
    commands: [
        {
            name: 'default',
            desc: '上传静态资源至快手cdn',
            usage: '[任意文件或文件夹]',
            example: [
                ['giao cdn <file>', '把当前路径的文件或者文件夹上传至快手cdn'],
                ['giao cdn', '没有指定上传文件的话，尝试上传剪贴板中的截图文件'],
            ],
            options: [
                {
                    key: 'hash',
                    type: OptionType.Boolean,
                    desc: '是否要给上传结果加上hash',
                    default: true,
                },
                {
                    key: 'internal',
                    alias: 'i',
                    type: OptionType.Boolean,
                    desc: '是否要上传内网cdn',
                    default: false,
                },
                {
                    key: 'delete',
                    alias: 'd',
                    type: OptionType.Boolean,
                    desc: '上传完成是否要删除源文件',
                    default: false,
                },
                {
                    key: 'limit',
                    type: OptionType.String,
                    desc:
                        '单次最大上传文件个数，强烈建议使用默认，除非有特殊需求（防止误操作把 home 或者 user 整个上传的情况）',
                },
            ],
        },
    ],
};

export default config;
